src/algorand_transactions/transactions_group.rs

```
@@ -0,0 +1,240 @@
+use serde::{Deserialize, Serialize};
+use sha2::Digest;
+
+use crate::{
+    algorand_errors::AlgorandError,
+    algorand_types::{Byte, Bytes, Result},
+    AlgorandHash,
+    AlgorandKeys,
+    AlgorandSignedTransaction,
+    AlgorandTransaction,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TxGroup {
+    #[serde(rename = "txlist", default)]
+    tx_group_hashes: Vec<AlgorandHash>,
+
+    #[serde(skip_serializing)]
+    txs: Vec<AlgorandTransaction>,
+
+    #[serde(skip_serializing)]
+    group_id: Option<AlgorandHash>,
+}
+
+impl TxGroup {
+    const MAX_TX_GROUP_SIZE: usize = 16;
+
+    pub fn new(transactions: &[AlgorandTransaction]) -> Result<TxGroup> {
+        let a = transactions.iter().map(|tx| tx.to_raw_tx_id()).collect();
+        match a {
+            Ok(res) => Ok(Self {
+                txs: transactions.to_vec(),
+                group_id: None,
+                tx_group_hashes: res,
+            }),
+            _ => Err(AlgorandError::Custom("error".to_string())),
```

Maybe it would be better to propagate the error: `Err(e) => Err(e)`

---

Yeah, this one needs changing for sure. See my review for more details (once I've finished it).